setup/terminal.md

```diff
@@ -50,30 +50,33 @@ nvm --help
 
 ### Text editor from terminal level 
 
-It is good to also know some basic text editors available from terminal level.
+It is good to know some basic text editors available from the terminal level.
 This is because your favorite graphical text editor might not always be installed 
 on someone else's computer while the terminal will always be there.
 Also if you work on a remote server machine the only thing which will be available to you will be the terminal.  
-The most popular one would be vi vim and nano. You do not have to be a super user and know all the shortcuts, 
-but it can be extremely useful if you can to basic operations like 
+The most popular one would be vi, vim and nano. You do not have to be a super user and know all the shortcuts,
+but it can be extremely useful if you can do basic operations like: 
 
 * create a new file
 * open existing file 
 * edit it 
 * save it
 
-Below examples how to do it in vim editor 
+Below are examples on how to do it in the vim editor
 
 To crete a new file or open and existing file
 
 ```
-vim mynewfile.txt
+vim myfile.txt
 ```
 
-To make an edition, press the letter "i" to switch the editor into editing mode
+The editor will start in the terminal. Vim editor has two `modes`of operation.
+Editor mode in which you can type the and edit the text and 
+Command mode where you can type commands to save, quit etc 
+By default it starts in operations mode, to switch to editor mode press the letter "i" 
 Move the cursor to the place you would like to edit and start typing 
 
-To save the changes press ESC then shift + semicolon key 
+To save the changes press ESC then shift + semicolon key to switch to command mode 
```

Comment:
To save the changes press ESC,
to switch to command mode press shift + semicolon key 

i think this is what you meant but if i'm wrong write it this way so its clearer to understand